### PR TITLE
Add hint to GNUPGHOME environment variable

### PR DESCRIPTION
**Pull request by justusbunsi**
With #19732, the default location for the `.gnupg` folder has changed. To mitigate this breaking change, users can specify the home directory for gnupg via `$GNUPGHOME` environment variable to keep using their current location.

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: docs/content/doc/advanced/signing.en-us.md [hunk 1]
```diff
@@ -100,7 +100,8 @@ ideal UI and therefore subject to change.
 **Since 1.17**, Gitea runs git in its own home directory `[repository].ROOT` and uses its own config `{[repository].ROOT}/.gitconfig`.
 If you have your own customized git config for Gitea, you should set these configs in system git config (aka `/etc/gitconfig`)
 or the Gitea internal git config `{[repository].ROOT}/.gitconfig`. 
-Related home files for git command (like `.gnupg`) should also be put in Gitea's git home directory `[repository].ROOT`. 
+Related home files for git command (like `.gnupg`) should also be put in Gitea's git home directory `[repository].ROOT`.\
```
Comment (wxiaoguang) [inline docs/content/doc/advanced/signing.en-us.md]:
```suggestion
Related home files for git command (like `.gnupg`) should also be put in Gitea's git home directory `[repository].ROOT`.
```

**Comment by justusbunsi** [comment]
Trimmed the explicit new line as suggested. I was not sure whether an explicit line break would help the readability.

**Comment by wxiaoguang** [comment]
> Trimmed the explicit new line as suggested. I was not sure whether an explicit line break would help the readability.

No worry in my mind. There will be no line break after rendering. There is no line break for the old document too.

image


image

**Comment by justusbunsi** [comment]
CI failure seems unrelated.

**Comment by zeripath** [comment]
Agh! this is a problem.

We shouldn't have set the repository root as the HOME directory.

This is not an appropriate directory especially as usernames can start with `.`

**Comment by wxiaoguang** [comment]
> Agh! this is a problem.

I do not think it's a problem.

> We shouldn't have set the repository root as the HOME directory.

`repository.ROOT` is the only directory can be used without breaking.

No other directory (created by Gitea) could be used for the Git HOME.

If we do not use `repository.ROOT` as Git HOME, then many users have to set an config option manually.

> This is not an appropriate directory especially as usernames can start with `.`

It won't. See the `IsUsableUsername`.

Directories starting with dot have special meaning, it's already a good practice across OS.

**Comment by wxiaoguang** [comment]
* https://github.com/go-gitea/gitea/pull/20136

**Comment by wxiaoguang** [comment]
Update: I over-thought something about the Git HOME problem.

Since it's fine to use a newly created directory in Gitea's data directory, there will a PR for it

**Review by lunny — APPROVED** [review]

**Comment by justusbunsi** [comment]
I guess #20114 makes the hint to that environment variable unrelated to 1.17 but still good to know for administrators. So I'll update this PR once the mentioned PR is done.

**Comment by wxiaoguang** [comment]
Let's wait for 
* https://github.com/go-gitea/gitea/pull/20114

Then the `{[repository].ROOT}` will be replaced by `{[git].HOME_PATH}` accordingly.

**Comment by lunny** [comment]
#20114 merged, please resolve the conflicts.

**Comment by justusbunsi** [comment]
Conflicts resolved.

**Comment by lunny** [comment]
make L-G-T-M work.

**Pull request merged**